src/schema/userTickets/mutations.ts

```diff
@@ -399,92 +398,92 @@ builder.mutationField("claimUserTicket", (t) =>
-                // error.
-                if (tickets.length + quantity > ticketTemplate.quantity) {
-                  throw new Error(
-                    `Not enough tickets for ticket template with id ${ticketId}`,
-                  );
-                }
-              }
-
               const isApproved =
                 ticketTemplate.isFree && !ticketTemplate.requiresApproval;
 
-              // If no errors were thrown, we can proceed to reserve the
-              // tickets.
-              const newTickets = new Array(quantity)
+              const newTickets = new Array(quantityToPurchase)
                 .fill(false)
                 .map(() => {
-                  const result = insertUserTicketsSchema.safeParse({
+                  const next: InsertUserTicketSchema = {
                     userId: USER.id,
                     purchaseOrderId: createdPurchaseOrder.id,
                     ticketTemplateId: ticketTemplate.id,
-                    paymentStatus: requiresPayment ? "unpaid" : "not_required",
                     approvalStatus: isApproved ? "approved" : "pending",
-                  });
+                  };
+
+                  return next;
+                });
 
-                  if (result.success) {
-                    return result.data;
-                  }
+              claimedTickets.push(...newTickets);
+            }
 
-                  logger.error("Could not parse user ticket", result.error);
-                })
-                .filter(Boolean);
+            for (const ticketTemplateId of ticketTemplatesIds) {
+              const newTickets = claimedTickets.filter(
+                (ticket) => ticket.ticketTemplateId === ticketTemplateId,
+              );
 
               logger.info(
-                `Creating ${newTickets.length} user tickets for ticket template with id ${ticketId}`,
-                { newTickets },
+                `Creating ${newTickets.length} user tickets for ticket template with id ${ticketTemplateId}`,
               );
+            }
 
-              if (newTickets.length === 0) {
-                throw new Error("Could not create user tickets");
-              }
+            // Bulk insert claimed tickets
+            const createdUserTickets = await trx
+              .insert(userTicketsSchema)
+              .values(claimedTickets)
+              .returning()
+              .execute();
+
+            // Bulk query for existing ticket counts
+            const finalTicketsCount = await trx
+              .select({
+                ticketTemplateId: userTicketsSchema.ticketTemplateId,
+                count: count(userTicketsSchema.id),
+              })
+              .from(userTicketsSchema)
+              .where(
+                and(
+                  inArray(
+                    userTicketsSchema.ticketTemplateId,
+                    ticketTemplates.map((t) => t.id),
+                  ),
+                  inArray(userTicketsSchema.approvalStatus, [
```

Comment:
Aquí me basé en el código anterior en el que se tomaban en cuenta solo estos dos estados, approved y pending, pero tal vez deberíamos tambien tomar en cuenta a:
- Gifted
- Gifted accepted
- Not required

¿Que piensan?